src/Microsoft.AspNetCore.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -365,6 +366,37 @@ public void Copy(Operation operation, object objectToApplyTo)
             }
         }
 
+        public void Test(Operation operation, object objectToApplyTo)
+        {
+            if (operation == null)
+            {
+                throw new ArgumentNullException(nameof(operation));
+            }
+
+            if (objectToApplyTo == null)
+            {
+                throw new ArgumentNullException(nameof(objectToApplyTo));
+            }
+
+            var parsedPath = new ParsedPath(operation.path);
+            var visitor = new ObjectVisitor(parsedPath, ContractResolver);
+
+            var target = objectToApplyTo;
+            if (!visitor.TryVisit(ref target, out var adapter, out var errorMessage))
+            {
+                var error = CreatePathNotFoundError(objectToApplyTo, operation.path, operation, errorMessage);
```

Comment:
Is this the right semantic for a `test`? it seems odd that you can't test against a non-existent value.

Reply:
If it cannot find that segment, how can it test against its value? Don't think the spec defines the exact behavior in that case.

https://tools.ietf.org/html/rfc6902#page-7

Reply:
Yeah the spec isn't super clear about it. This is the closest thing I can find: https://tools.ietf.org/html/rfc6902#appendix-A.12

> A.12.  Adding to a Nonexistent Target
>
>   An example target JSON document:
>
>   { "foo": "bar" }
>
>   A JSON Patch document:
>
>   [
     { "op": "add", "path": "/baz/bat", "value": "qux" }
   ]
>
>   This JSON Patch document, applied to the target JSON document above,
   would result in an error (therefore, it would not be applied),
   because the "add" operation's target location that references neither
   the root of the document, nor a member of an existing object, nor a
   member of an existing array.

Which would lead me to believe that much like doing an `add` to something that doesn't exist is a failure (i.e. abort the whole operation), then doing a `test` in a similar case would also be a failure. (Here's where failures are discussed: https://tools.ietf.org/html/rfc6902#section-5.)

Reply:
Yeah, the section about test talks about comparing values. I think they mean to say that there must be a value present in the target - but I would like it better if it was spelled out.